### PR TITLE
fix: flag hidden elms with disallowed role(s) for review

### PR DESCRIPTION
**Pull request by jkodu**
This PR does the below for the rule `aria-allowed-role`

- Note: The rule spec does not exclude hidden elements `excludeHidden: false` (this is by choice).
- Change has been put in to flag elements as `needs review`, which are hidden (via CSS and to screen readers), and have redundant/ disallowed roles.

Closes issue:
- https://github.com/dequelabs/axe-core/issues/1194

## Reviewer checks

**Required fields, to be filled out by PR reviewer(s)**
- [x] Follows the commit message policy, appropriate for next version
- [x] Has documentation updated, a DU ticket, or requires no documentation change
- [x] Includes new tests, or was unnecessary
- [x] Code is reviewed for security by @WilcoFiers

### CODE REVIEW

File: lib/checks/aria/aria-allowed-role.js [hunk 1]
```diff
@@ -18,6 +19,10 @@ const unallowedRoles = axe.commons.aria.getElementUnallowedRoles(
 
 if (unallowedRoles.length) {
 	this.data(unallowedRoles);
+	if (!dom.isVisible(node, true)) {
+		// flag hidden elements for review
+		return null;
```
Comment (WilcoFiers) [inline lib/checks/aria/aria-allowed-role.js]: This should return undefined. I know this also works, but we should do this consistently.

File: lib/checks/aria/aria-allowed-role.js [hunk 2]
```diff
@@ -3,6 +3,7 @@
  * https://www.w3.org/TR/html-aria/#docconformance
  * https://www.w3.org/TR/SVG2/struct.html#implicit-aria-semantics
  */
+const { dom } = axe.commons;
```
Comment (WilcoFiers) [inline lib/checks/aria/aria-allowed-role.js]: Please add an "inapplicable" message to the check JSON so that we communicate with the user what's going on.

Comment (jkodu) [reply lib/checks/aria/aria-allowed-role.js]: @WilcoFiers 

I have gone with below. Let me know if you have suggestions.

`"incomplete": "Element is not visible, and ARIA role{{=it.data && it.data.length > 1 ? 's' : ''}} {{=it.data.join(', ')}} {{=it.data && it.data.length > 1 ? 'are' : ' is'}} not allowed for given element"`

Comment (jkodu) [reply lib/checks/aria/aria-allowed-role.js]: @marcysutton - any suggestions for the incomplete message - https://github.com/dequelabs/axe-core/pull/1225/files#diff-2f072a5102050cd26c55026d5d9f9c4fR13

File: test/checks/aria/aria-allowed-role.js [hunk 3]
```diff
@@ -51,6 +51,32 @@ describe('aria-allowed-role', function() {
 		);
 	});
 
+	it('returns undefined (needs review) when element is hidden and has redundant role', function() {
```
Comment (WilcoFiers) [inline test/checks/aria/aria-allowed-role.js]: This is not a "redundant role".

File: test/checks/aria/aria-allowed-role.js [hunk 4]
```diff
@@ -51,6 +51,32 @@ describe('aria-allowed-role', function() {
...
+			'role="presentation"></button>';
+		var target = fixture.querySelector('#target');
+		var actual = checks['aria-allowed-role'].evaluate.call(
+			checkContext,
+			target
+		);
+		assert.isUndefined(actual);
+	});
+
+	it('returns undefined (needs review) when element is with in hidden parent and has redundant role', function() {
```
Comment (WilcoFiers) [inline test/checks/aria/aria-allowed-role.js]: Same here, not a "redundant role".

**Pull request merged**